language-guides/spring-data-neo4j/spring-data-neo4j.adoc

```diff
@@ -34,129 +35,194 @@ When developing with Neo4j, please use JDK 8 and your favorite IDE.
+The direction property is outgoing by default, so we must specify that the relationship is incoming on the `Movie` node.
+To capture additional attributes about the relationship between `Person` and `Movie`, a new entity called `Role` is created for `Person` to be connected to `Movie` by the `Role` he/she played.
+In this class, we annotate it as a relationship entity and specify the name of the relationship (`ACTED_IN`).
+The `Role` class also has an `ID` field that is generated by the database and a list to contain the possible roles that a person could play in a movie (could be more than one).
+Finally, annotations are added to mark the `Person` and `Movie` nodes as start and end nodes for the relationship.
+We now have our graph structure mapped in our application. This is the object-graph mapping (OGM) piece.
 
 .Entities
 [source,java]
 ----
+@NodeEntity
+public class Person {
+   
+   @Id @GeneratedValue private Long id;
+   private String name;
+   private int born;
+
+   @Relationship(type = "ACTED_IN")
+   private List<Movie> movies = new ArrayList<>();
+}
+
 @NodeEntity
 public class Movie {
+   
+   @Id @GeneratedValue private Long id;
+   private String title;
+   private int released;
+   private String tagline;
+
+   @JsonIgnoreProperties("movie")
+   @Relationship(type = "ACTED_IN", direction = Relationship.INCOMING)
+   private List<Role> roles;
+}
 
-  @Id @GeneratedValue Long id;
+@RelationshipEntity(type = "ACTED_IN")
+public class Role {
 
-  String title;
+    @Id @GeneratedValue private Long id;
+	private List<String> roles = new ArrayList<>();
 
-  Person director;
+	@StartNode
+	private Person person;
 
-  @Relationship(type="ACTED_IN", direction = Relationship.INCOMING)
-  Set<Person> actors = new HashSet<>();
-}
+	@EndNode
+	private Movie movie;
 ----
 
+The repository interface allows the developer to create methods and queries needed for retrieving the data from the database.
+The interface extends the `Neo4jRepository`, which extends the Spring CRUDRepository for persisting and retrieving data.
+Some queries can be derived for us (like the `findByTitle` method), but others must be specifically written and annotated with `@Query`.
+
+There are also a lot of other methods available by default, such as `findOne`, `findAll`, etc.
+For more information, see the Spring Data Neo4j page and choose the latest https://projects.spring.io/spring-data-neo4j/[documentation link^] on the right side of the page.
+
 .Declare a repository interface
 [source,java]
 ----
-interface MovieRepository extends Neo4jRepository<Movie, Long> {
+public interface MovieRepository extends Neo4jRepository<Movie, Long> {
 
-  @Query("MATCH (m:Movie)<-[rating:RATED]-(user)
-          WHERE id(m) = {movieId} RETURN rating")
-  List<Rating> getRatings(@Param("movieID") Long movieId);
+    Movie findByTitle(@Param("title") String title);
```

Comment:
somehow I still feel the two `findByTitle*` methods don't cut it for the reader ... ?
Perhaps just leave one off to not confuse them?